lib/replicator.js

```diff
@@ -2461,6 +2461,8 @@ module.exports = class Replicator {
   }
 
   _makePeer (protomux) {
+    if (!this.downloading) return false
```

Comment:
This needs some thought, because I think it could be dangerous. Doesn't this mean we will never open replication sessions to peers if we're not downloading the hypercore ourselves?

So the underlying assumption is: we always have `this.downloading` set to `true` when we open a connection and actually want to replicate.

For example, this assumes there's no path where we open a connection to a peer while `this.downloading === false` but set it to true later.

(not saying this isn't correct, it just rings some alarm bells of 'potentially dangerous'. The fact that all tests pass does seem like a good sign)

Reply:
ya true, we should add a test that checks if an inactive peer can replicate with an active peer if we dont